From 80d98c6ebbddaccc1a80a849615c70f4bd7cc2ac Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 24 Jan 2022 14:30:53 +0200
Subject: [PATCH] [SW-58071] gaudi3: clear status register

As part of ROM init flow, this patch add change to clear status
registers.

Change-Id: Ia341de4cbd6117c4a765acdc1afa8a54e45e8f83
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index c7bba15..978c9fb 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -6,7 +6,21 @@
 
 #include <arcboot.h>
 
-void arcboot_rom_clear_sw_status(void) {}
+void arcboot_rom_clear_sw_status(void)
+{
+	/* As per Architecture recommendation, clear status reg */
+	regmap_writel(0, mmD0_PSOC_SECURITY_STATUS);
+
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_0);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_1);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_2);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_3);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_4);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_5);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_6);
+	regmap_writel(0, mmD0_PSOC_SECURITY_SW_STATUS_7);
+}
+
 void update_warm_reset_config(void) {}
 void arcboot_rom_set_spi_write_limit(void) {}
 void pcie_phy_para_sel_set(void) {}
@@ -42,4 +56,7 @@
 void arcboot_rom_init(void)
 {
 	arcboot_rom_global.die_id = regmap_readl(mmD0_PARC_LOCAL_CFG_DIE_ID);
+
+	/* clear SW Status registers */
+	arcboot_rom_clear_sw_status();
 }
